CRM/Admin/Form/Setting.php

```diff
@@ -96,9 +97,13 @@ public function buildQuickForm() {
     );
 
     $descriptions = array();
-    foreach ($this->_settings as $setting => $group) {
-      $settingMetaData = civicrm_api('setting', 'getfields', array('version' => 3, 'name' => $setting));
-      $props = $settingMetaData['values'][$setting];
+    // Getting all once seems better than multiple calls.
+    $allSettingMetaData = civicrm_api3('setting', 'getfields', array());
+    $settingMetaData = array_intersect_key($allSettingMetaData['values'], $this->_settings);
+    // This array_merge re-orders to the key order of $this->_settings.
+    $settingMetaData = array_merge($this->_settings, $settingMetaData);
+
+    foreach ($settingMetaData as $setting => $props) {
```

Comment:
actually would be nicer (I think( if you moved the whole loop to another separate private method such buildingSettingFields($settingMetaData) or something like that  .. it would make the could much easier to understand